public/app/plugins/datasource/tempo/streaming.ts

```diff
@@ -71,11 +71,16 @@ export function doTempoSearchStreaming(
         if ('message' in evt && evt?.message) {
           const currentTime = performance.now();
           const elapsedTime = currentTime - requestTime;
-          // Schema should be [traces, metrics, state, error]
-          const traces = evt.message.data.values[0][0];
-          const metrics = evt.message.data.values[1][0];
-          const frameState: SearchStreamingState = evt.message.data.values[2][0];
-          const error = evt.message.data.values[3][0];
+
+          const dataIndex = evt.message.schema.fields.findIndex((f: DataFrame) => f.name === 'result');
```

Comment:
I think you can also create a [FieldCache](https://github.com/grafana/grafana/blob/v11.5.0/packages/grafana-data/src/dataframe/FieldCache.ts#L9) to avoid looping over fields multiple times